lib/ClangImporter/ImportDecl.cpp

```diff
@@ -6757,9 +6757,8 @@ SwiftDeclConverter::importSubscript(Decl *decl,
   // - A getter is redeclared in a subclass, but not the setter.
   // - The getter and setter are part of the same type.
   // - There is no setter.
-  bool associateWithSetter = !getterAndSetterInSameType && setter == decl;
   DeclContext *dc =
-      associateWithSetter ? setter->getDeclContext() : getter->getDeclContext();
+      !getterAndSetterInSameType && setter == decl ? setter->getDeclContext() : getter->getDeclContext();
```

Comment:
Sorry, what I meant before was that `setter` would never be null if `associateWithSetter` was false, which meant `setter->getDeclContext()` is perfectly fine to do. The null check is not needed.

I don't think this improves anything as it's just inlining the `associateWithSetter` computation.

Reply:
Closing this.